sympy/core/symbol.py

```diff
@@ -314,7 +314,46 @@ def free_symbols(self):
     binary_symbols = free_symbols  # in this case, not always
 
     def as_set(self):
-        return S.UniversalSet
+        from sympy.sets import FiniteSet, Intersection, Interval
```

Comment:
I think that `as_set` should have some explanation of what it does (i.e. a docstring).

In particular when I wrote the initial mock up of this code it was in the context of trying to helpfully raise an error when the assumptions are invalid so it seemed find to have `Symbol.as_set` return a possible superset of the possibilities.

We need to clarify what the guarantee here is so if `x` is a symbol and `X` is the set returned by `x.as_set()` which of these are we saying holds?

1. Any possible value for `x` that is consistent with the assumptions on `x` will be an element of `X`.
2. Any element of `X` is a possible value for `x` consistent with the assumptions on `x`.

If *both* of these are true then we can say for sure that `Eq(z, x)` is equivalent to `Contains(z, X)` in all cases. Otherwise we only have a unidirectional implication. That's fine if it's just to warn about invalid use in solveset but the meaning or definition of the set returned should be clarified.

There is one case (primes) that raises. If we want both propositions above to hold then we can try to ensure that we raise for any assumption we don't know how to handle but that means we do need to check them all e.g. `algebraic`.

Reply:
I am not sure how to check other implications. I just know that in the case of prime there are not enough assumptions to check whether a symbol is prime or not.

The only place this is being used (at present) in in `solveset`.

Reply:
I removed this but it is in the history and in the original issue that referenced this.